runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/internal/catalog/TestCatalogHandler.java

```diff
@@ -15,23 +15,61 @@
  */
 package io.aklivity.zilla.runtime.engine.test.internal.catalog;
 
+import java.nio.ByteOrder;
+
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+
 import io.aklivity.zilla.runtime.engine.catalog.CatalogHandler;
 import io.aklivity.zilla.runtime.engine.test.internal.catalog.config.TestCatalogOptionsConfig;
+import io.aklivity.zilla.runtime.engine.validator.function.ValueConsumer;
 
 public class TestCatalogHandler implements CatalogHandler
 {
+    private static final int MAX_PADDING_LEN = 10;
+    private static final byte MAGIC_BYTE = 0x0;
+    private static final int ENRICHED_LENGTH = 5;
+
     private final String schema;
+    private final MutableDirectBuffer prefixRO;
+    private final int id;
+    private final boolean embed;
+    private final boolean exclude;
 
     public TestCatalogHandler(
-        TestCatalogOptionsConfig options)
+        TestCatalogOptionsConfig config)
+    {
+        this.schema = config.schema;
+        this.prefixRO = new UnsafeBuffer(new byte[5]);
+        this.id = config.id;
+        this.embed = config.embed;
+        this.exclude = config.exclude;
+    }
+
+    @Override
+    public int enrich(
+        int schemaId,
+        ValueConsumer next)
     {
-        this.schema = options.schema;
+        int length = 0;
+        if (embed)
+        {
+            prefixRO.putByte(0, MAGIC_BYTE);
+            prefixRO.putInt(1, schemaId, ByteOrder.BIG_ENDIAN);
+            next.accept(prefixRO, 0, 5);
+            length = ENRICHED_LENGTH;
+        }
+        else if (exclude)
+        {
+            length = ENRICHED_LENGTH;
+        }
+        return length;
```

Comment:
Does it make sense to have booleans for `embed` and `exclude` or is this more a funciton of whether the validator is read vs write?

If it is a property of the validator, then perhaps we need to pass this context from the validator to the catalog handler, either via a parameter or by having 2 methods on catalog handler?